### PR TITLE
[New Rule] Backup Files Deletion

### LINKED ISSUE #1515: [New Rule] Backup Files Deletion
## Description

Identifies the deletion of Backup Files.

EQL

```
file where event.type == "deletion" and
  (
  /* Veeam Related Backup Files */
  (file.extension : ("VBK", "VIB", "VBM") and
  not process.executable : ("?:\\Windows\\Veeam\\Backup\\*", "?:\\Program Files\\Veeam\\Backup and Replication\\*")) or

  /* Veritas Backup Exec Related Backup File */
  (file.extension : ("BKF") and
  not process.executable : ("?:\\Program Files\\Veritas\\Backup Exec\\*"))
  )
```

### Target Operating Systems
Windows

### Tested ECS Version
x.x.x

### References
https://www.advintel.io/post/backup-removal-solutions-from-conti-ransomware-with-love

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Resolves #1515

### CODE REVIEW

File: rules/windows/impact_backup_file_deletion.toml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+query = '''
+file where event.type == "deletion" and
+  (
+  /* Veeam Related Backup Files */
+  (file.extension : ("VBK", "VIB", "VBM") and
+  not process.executable : ("?:\\Windows\\Veeam\\Backup\\*", "?:\\Program Files\\Veeam\\Backup and Replication\\*")) or
+
+  /* Veritas Backup Exec Related Backup File */
+  (file.extension : ("BKF") and
+  not process.executable : ("?:\\Program Files\\Veritas\\Backup Exec\\*"))
```
Comment (Samirbous) [inline rules/windows/impact_backup_file_deletion.toml:34]: `file.extension == "BKF"` (no need for parentheses) , maybe also add the x86 equivalent path  `?:\\Program Files (x86)\\Veritas\\*` ,  `"?:\\Program Files (x86)\\Veeam\\*"`.

File: rules/windows/impact_backup_file_deletion.toml [hunk 2]
```diff
@@ -0,0 +1,51 @@
+[metadata]
+creation_date = "2021/10/01"
+maturity = "production"
+updated_date = "2021/10/01"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the deletion of Backup Files. Adversaries may delete Backup files
+to ensure that recovery from a Ransomware attack is less likely.
```
Comment (Samirbous) [inline rules/windows/impact_backup_file_deletion.toml:10]: maybe need to be linted (one line)

**Review by Samirbous — APPROVED** [review]
nice rule, left some suggestions.

File: rules/windows/impact_backup_file_deletion.toml (hunk 1 above)
Comment (w0rk3r) [reply rules/windows/impact_backup_file_deletion.toml]: Done, thanks!

File: rules/windows/impact_backup_file_deletion.toml (hunk 2 above)
Comment (w0rk3r) [reply rules/windows/impact_backup_file_deletion.toml]: I've added some more details, to the description

File: rules/windows/impact_backup_file_deletion.toml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+author = ["Elastic"]
+description = """
+Identifies the deletion of Backup Files done by used other ways than using the backup solution.
+Adversaries may delete Backup files to ensure that recovery from a Ransomware attack is less likely.
+"""
+from = "now-9m"
+index = ["winlogbeat-*", "logs-endpoint.events.*", "logs-windows.*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Backup Files Deletion"
```
Comment (brokensound77) [inline rules/windows/impact_backup_file_deletion.toml:16]:
```suggestion
name = "Third-party Backup Files Deleted via Unexpected Process"
```

File: rules/windows/impact_backup_file_deletion.toml [hunk 4]
```diff
@@ -0,0 +1,54 @@
+[metadata]
+creation_date = "2021/10/01"
+maturity = "production"
+updated_date = "2021/10/01"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies the deletion of Backup Files done by used other ways than using the backup solution.
+Adversaries may delete Backup files to ensure that recovery from a Ransomware attack is less likely.
```
Comment (brokensound77) [inline rules/windows/impact_backup_file_deletion.toml:10]:
```suggestion
description = """Identifies the deletion of backup files, saved using third-party software, by a process outside of the backup suite.
Adversaries may delete Backup files to ensure that recovery from a Ransomware attack is less likely or for the sake of data destruction.
```

Need to re-lint this (`python -m detection_rules toml-lint -f <file>`) or just manually break at 120 lines

**Review by brokensound77 — COMMENTED** [review]
Can you add an entry under `false_positives` for potential expected activity. 

Also left a few comments

**Review by brokensound77 — APPROVED** [review]
LGTM 👍 , nice job!

**Pull request merged**